packages/atomic/src/components/search/atomic-query-summary/atomic-query-summary.spec.ts

```diff
@@ -0,0 +1,373 @@
+import {buildQuerySummary, type QuerySummaryState} from '@coveo/headless';
+import {html} from 'lit';
+import {describe, expect, it, vi} from 'vitest';
+import {renderInAtomicSearchInterface} from '@/vitest-utils/testing-helpers/fixtures/atomic/search/atomic-search-interface-fixture';
+import {buildFakeSearchEngine} from '@/vitest-utils/testing-helpers/fixtures/headless/search/engine';
+import {buildFakeSummary} from '@/vitest-utils/testing-helpers/fixtures/headless/search/summary-subcontroller';
+import './atomic-query-summary';
+import {AriaLiveRegionController} from '@/src/utils/accessibility-utils';
+import type {AtomicQuerySummary} from './atomic-query-summary';
+
+vi.mock('@coveo/headless', {spy: true});
+
+describe('atomic-query-summary', () => {
+  const mockedEngine = buildFakeSearchEngine();
+  let mockedQuerySummary: ReturnType<typeof buildQuerySummary>;
+
+  const renderQuerySummary = async ({
+    querySummaryState = {},
+  }: {
+    querySummaryState?: Partial<QuerySummaryState>;
+  } = {}) => {
+    mockedQuerySummary = buildFakeSummary({state: querySummaryState});
+
+    vi.mocked(buildQuerySummary).mockReturnValue(mockedQuerySummary);
+
+    const {element} = await renderInAtomicSearchInterface<AtomicQuerySummary>({
+      template: html`<atomic-query-summary></atomic-query-summary>`,
+      selector: 'atomic-query-summary',
+      bindings: (bindings) => {
+        bindings.engine = mockedEngine;
+        return bindings;
+      },
+    });
+
+    return {
+      element,
+      placeholder: element.shadowRoot!.querySelector('[part="placeholder"]'),
+      container: element.shadowRoot!.querySelector('[part="container"]'),
+      parts: (element: AtomicQuerySummary) => {
+        const qs = (part: string) =>
+          element.shadowRoot?.querySelector(`[part*="${part}"]`);
+        return {
+          container: qs('container'),
+          highlight: qs('highlight'),
+          query: qs('query'),
+          placeholder: qs('placeholder'),
+          duration: qs('duration'),
+        };
+      },
+    };
+  };
+
+  // Guard condition tests (outside the rendering context)
```

Comment:
```suggestion
```

This comment is misleading; you are actually testing the guard in the rendering context, it's just that you're expecting nothing to be rendered 😁